DentOS_Framework/DentOsTestbedLib/src/dent_os_testbed/lib/traffic/ixnetwork/ixnetwork_ixia_client_impl.py

```diff
@@ -518,10 +518,18 @@ def run_traffic_item(self, device, command, *argv, **kwarg):
             device.applog.info("Clear Stats")
             IxnetworkIxiaClientImpl.ixnet.ClearStats()
         elif command == "clear_traffic":
-            for traffic_item in IxnetworkIxiaClientImpl.tis:
-                device.applog.info(f"TI to be removed: {traffic_item.Name}")
-                traffic_item.remove()
-            IxnetworkIxiaClientImpl.tis.clear()
+            traffic_names = kwarg["params"][0]["traffic_names"]
+            if traffic_names:
+                for traffic_item in IxnetworkIxiaClientImpl.tis[:]:
+                    if traffic_item.Name in traffic_names:
+                        device.applog.info(f"TI to be removed: {traffic_item.Name}")
+                        traffic_item.remove()
+                        IxnetworkIxiaClientImpl.tis.remove(traffic_item)
+            else:
+                for traffic_item in IxnetworkIxiaClientImpl.tis:
+                    device.applog.info(f"TI to be removed: {traffic_item.Name}")
+                    traffic_item.remove()
+                IxnetworkIxiaClientImpl.tis.clear()
```

Comment:
```suggestion
            tis_to_remove = IxnetworkIxiaClientImpl.tis
            if traffic_names:
                tis_to_remove = filter(lambda ti: ti.Name in traffic_names, IxnetworkIxiaClientImpl.tis)
            for traffic_item in tis_to_remove:
                device.applog.info(f"TI to be removed: {traffic_item.Name}")
                traffic_item.remove()
            IxnetworkIxiaClientImpl.tis.clear()
```

Reply:
The change that you are suggesting has a bug in it. We don't want to clear the `tis` list unconditionally, but only when we are removing all traffic items. It's better to change `tis.clear()` to `tis.remove(traffic_item)`.